flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/BatchingStateChangeUploader.java

```diff
@@ -206,16 +251,27 @@ private synchronized void setErrorSafe(Throwable t) {
     }
 
     private static UploadTask wrapWithSizeUpdate(
-            UploadTask uploadTask, long preComputedTaskSize, LongAdder inflightSize) {
+            UploadTask uploadTask, Runnable releaseCapacityCallback) {
         return new UploadTask(
                 uploadTask.changeSets,
                 result -> {
-                    inflightSize.add(-preComputedTaskSize);
-                    uploadTask.successCallback.accept(result);
+                    try {
+                        releaseCapacityCallback.run();
+                    } finally {
+                        uploadTask.successCallback.accept(result);
+                    }
                 },
                 (result, error) -> {
-                    inflightSize.add(-preComputedTaskSize);
-                    uploadTask.failureCallback.accept(result, error);
+                    try {
+                        releaseCapacityCallback.run();
+                    } finally {
+                        uploadTask.failureCallback.accept(result, error);
+                    }
                 });
     }
+
+    @Override
+    public AvailabilityProvider getAvailabilityProvider() {
+        return availabilityHelper;
+    }
```

Comment:
Can you include here in a comment the explanation why we think it's fine to have this unsynchronised, despite that other task threads might actually make this unavailable?